tq/basic_upload.go

```diff
@@ -74,6 +72,25 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 	}
 	defer f.Close()
 
+	if len(req.Header.Get("Content-Type")) == 0 {
+		buffer := make([]byte, 512)
+		n, err := f.Read(buffer)
+		if err != nil && err != io.EOF {
+			return errors.Wrap(err, "basic upload")
+		}
+
+		contentType := http.DetectContentType(buffer[:n])
+		if _, err := f.Seek(0, 0); err != nil {
+			return errors.Wrap(err, "basic upload")
```

Comment:
I have the same thought about this error wrap as I do the above, maybe replacing "content type detect" with "content type rewind" instead.